generators/app/templates/src/containers/Profile/profile.container.js

```diff
@@ -0,0 +1,215 @@
+   * and will add updated flag to true, this will be taken
+   * for submit form to check which fields needs to be sent to POD
+  */
+  onInputChange = (e: Event) => {
+    const name = e.target.name;
+    const value = e.target.value;
+
+    const updatedFormField = this.state.formFields.map(field => {
+      if (field.property === name || field.blankNode === name) {
+        return {
+          ...field,
+          updated: true,
+          value
+        };
+      }
+
+      return { ...field };
+    });
+
+    this.setState({ formFields: updatedFormField });
+  };
+  /**
+   * onSubmit will send all the updated fields to POD
+   * fields that was not updated will be not send it.
+  */
+  onSubmit = async (e: Event) => {
+    try {
+      e.preventDefault();
+      let node;
+      const updatedFormField = await Promise.all(
+        this.state.formFields.map(async field => {
+          if (field.updated) {
+            node = data.user[field.property];
+            if (field.blankNode) {
+              node = data[field.nodeParentUri][field.blankNode];
+            }
+            await node.set(field.value);
+
+            return {
+              ...field,
+              updated: false
+            };
+          }
+          return { ...field };
+        })
+      );
+      this.props.toastManager.add("Profile was updated successfully", {
+        appearance: "success"
+      });
+      this.setState({ formFields: updatedFormField });
+    } catch (error) {
+      this.props.toastManager.add(error.message, { appearance: "error" });
+    }
+  };
+  /**
+   * Fetch profile photo from card
+   */
+  fetchPhoto = async () => {
+    try {
+      // We are fetching profile card document
+      const user = data[this.props.webId];
+      // We access to document node using a node name
+      let image = await user.image;
+      let hasImage = true;
+      // If image is not present on card we try with hasPhoto
+      if (!image) {
+        /**
+         * hasPhoto is a new context that ldflex doesn't having
+         * we need to add it manually.
+         * if you want to know more about context please go to:
+         * https://github.com/digitalbazaar/jsonld.js
+         */
+        image = await user["vcard:hasPhoto"];
+
+        hasImage = false;
+      }
+
+      this.setState({
+        photo: (image && image.value) || defaulProfilePhoto,
+        hasImage
+      });
+    } catch (error) {
+      this.props.toastManager.add(error.message, { appearance: "error" });
+    }
+  };
+  /**
+   * updatedPhoto will update the photo url on vcard file
+   * this function will check if user has image or hasPhoto node if not
+   * will just update it, the idea is use image instead of hasPhoto
+   * @params{String} uri photo url
+  */
+  updatePhoto = async (uri: String) => {
+    try {
+      const { user } = data;
+      this.state.hasImage
+        ? await user.image.set(uri)
+        : await user.image.add(uri);
```

Comment:
In both cases, `set` is supposed to work, I think. @rubensworks?

Reply:
(this not working _might_ be a node-solid-server bug)

Reply:
Hi @RubenVerborgh, yes we have a conflict issue when node not exists on the document, for example image if this node is not on the vcard and try to set it will show a conflict error.

Reply:
Yes, do keep it like this; that might be a server bug (https://github.com/solid/node-solid-server/issues/1085)